cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -315,44 +315,53 @@ def _add_resources(self):
                 cw_log_group=self.log_group,
             )
 
-            self._add_alarms()
+            self._add_head_node_alarms()
+
+    def _cw_metric_head_node(self, namespace, metric_name, statistic="Maximum", period_seconds=CW_ALARM_PERIOD_DEFAULT):
+        return cloudwatch.Metric(
+            namespace=namespace,
+            metric_name=metric_name,
+            dimensions_map={"InstanceId": self.head_node_instance.ref},
+            statistic=statistic,
+            period=Duration.seconds(period_seconds),
+        )
 
-    def _add_alarms(self):
-        self.alarms = []
+    def _add_head_node_alarms(self):
+        self.head_node_alarms = []
 
         metrics_for_alarms = {
-            "Mem": cloudwatch.Metric(
-                namespace="CWAgent",
-                metric_name="mem_used_percent",
-                dimensions_map={"InstanceId": self.head_node_instance.ref},
-                statistic="Maximum",
-                period=Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
-            ),
-            "Disk": cloudwatch.Metric(
-                namespace="CWAgent",
-                metric_name="disk_used_percent",
-                dimensions_map={"InstanceId": self.head_node_instance.ref, "path": "/"},
-                statistic="Maximum",
-                period=Duration.seconds(CW_ALARM_PERIOD_DEFAULT),
-            ),
+            "Health": self._cw_metric_head_node("AWS/EC2", "StatusCheckFailed"),
+            "Cpu": self._cw_metric_head_node("AWS/EC2", "CPUUtilization"),
+            "Mem": self._cw_metric_head_node("CWAgent", "mem_used_percent"),
+            "Disk": self._cw_metric_head_node("CWAgent", "disk_used_percent"),
```

Comment:
unfortunately dimension_map is different for Disk, we need to set a default parameter for others and pass `dimensions_map={"InstanceId": self.head_node_instance.ref, "path": "/"}` for disk metric

Reply:
yes, just addressed but without increasign code dup